src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -100,25 +100,61 @@ public List<AttachmentUnit> splitAndSaveUnits(LectureUnitInformationDTO lectureU
         }
     }
 
+    /**
+     * Gets the slides that should be removed by the given keyphrase
+     *
+     * @param fileBytes                The byte content of the file (lecture slides) to be split
+     * @param commaSeparatedKeyPhrases key phrases that identify slides about to be removed
+     * @return list of the number of slides that will be removed
+     */
+    public List<Integer> getSlidesToRemoveByKeyphrase(byte[] fileBytes, String commaSeparatedKeyPhrases) {
+        List<Integer> slidesToRemove = new ArrayList<>();
+        if (commaSeparatedKeyPhrases.isEmpty()) {
+            return slidesToRemove;
+        }
+        try (PDDocument document = Loader.loadPDF(fileBytes)) {
+            PDFTextStripper pdfTextStripper = new PDFTextStripper();
+            Splitter pdfSplitter = new Splitter();
+            List<PDDocument> pages = pdfSplitter.split(document);
+            List<String> keyphrasesList = Arrays.stream(commaSeparatedKeyPhrases.split(",")).filter(s -> !s.trim().isEmpty()).toList();
```

Comment:
This logic is currently duplicated, please move this into its own method.

Then, you could also allow some form of escaping to allow the usage of commas inside of the keyphrase.

Reply:
I added the Splitter and Stripper as class variables as they are used in even more places. I also added a method to get the list of keyprhases (so it can be easily modified in the future). 

IMO it makes little sense to allow commas in Keyphrases. This would only help in very rare edge cases, while being more confusing for the average user.
E.g. you would need a keyphrase like "Hello, dear Students" and other slides containing "Hello" and "dear Students" that you don't want removed. (Also if such slides are in-between chapters they can still be excluded from the chapters anyways)